sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -382,7 +464,25 @@ static az_result disconnecting(az_event_policy* me, az_event event)
   switch (event.type)
   {
     case AZ_HFSM_EVENT_ENTRY:
+      _az_RETURN_IF_FAILED(az_platform_timer_create(
+          &this_policy->_internal.connection_timer, _connection_timeout_callback, this_policy));
+      _az_RETURN_IF_FAILED(az_platform_timer_start(
+          &this_policy->_internal.connection_timer,
+          this_policy->_internal.disconnecting_timeout_msec));
+      break;
     case AZ_HFSM_EVENT_EXIT:
```

Comment:
Need to add timeout state here

Reply:
The timeout event is already handled by the super-state (Started) which causes a state change to Faulted. The reason is that the network is left in an uncertain state.